### PR TITLE
remove inefficient maildirwatch

### PR DESCRIPTION
**Pull request by DerDakon**
This made sense at a time where maildir was a new concept, but most relevant mailers natively support maildir these days. There are also more efficient syscalls to watch directories meanwhile, and people usually also do not have local mail stores anyway.

### CODE REVIEW

**Comment by schmonz** [comment]
> we should publish our intent in advance

Second (maybe third) thoughts: meh. Maybe that'll be important when we have more users. For now, if we want to delete, let's delete.

File: CHANGES [hunk 1]
```diff
@@ -1,3 +1,4 @@
+20190827 remove maildirwatch, you can do this more efficiently using *notify() syscalls
```
Comment (schmonz) [inline CHANGES]: maybe just "remove inefficient maildirwatch"?

Comment (DerDakon) [reply CHANGES]: Yes, the rest could be in the commit message.

**Comment by schmonz** [comment]
> > we should publish our intent in advance
> 
> Second (maybe third) thoughts: meh. Maybe that'll be important when we have more users. For now, if we want to delete, let's delete.

Fourth thoughts, after IRC discussion with @alanpost: Given our claim to be conservative and careful, if we remove a program that someone was relying on without giving them a chance to change our minds, they will consider our claim false. It's expensive to have users perceiving us as rash; it's cheap to wait a little longer before deleting (or treating as extensions, which might feel pretty similar).

**Comment by DerDakon** [comment]
fine with me.

**Pull request merged**